scripts/parse_lake_manifest_changes.py

```diff
@@ -0,0 +1,126 @@
+    new_packages = {pkg['name']: pkg for pkg in new_manifest['packages']}
+
+    # Find packages with different revisions and new packages
+    for name, new_pkg in new_packages.items():
+        if name in old_packages:
+            old_pkg = old_packages[name]
+            if old_pkg['rev'] != new_pkg['rev']:
+                changes.append({
+                    'type': 'update',
+                    'dependency': name,
+                    'old_version': old_pkg['rev'],
+                    'new_version': new_pkg['rev'],
+                    'url': new_pkg['url']
+                })
+        else:
+            changes.append({
+                'type': 'add',
+                'dependency': name,
+                'version': new_pkg['rev'],
+                'url': new_pkg['url']
+            })
+
+    # Find removed packages
+    for name in old_packages:
+        if name not in new_packages:
+            changes.append({
+                'type': 'remove',
+                'dependency': name,
+                'version': old_packages[name]['rev'],
+                'url': old_packages[name]['url']
+            })
+
+    return changes
+
+def get_github_diff_url(pkg_url, old_rev, new_rev):
+    """Convert a package URL to a GitHub compare URL"""
+    # For updates: compare URL
+    if old_rev and new_rev:
+        return f"{pkg_url}/compare/{old_rev}...{new_rev}"
+    # For additions: single new commit URL
+    elif new_rev is not None and old_rev is None:
+        return f"{pkg_url}/commit/{new_rev}"
+    # For removals: single old commit URL
+    elif old_rev is not None and new_rev is None:
+        return f"{pkg_url}/commit/{old_rev}"
+    # Handle case where no revisions are provided
+    else:
+        return pkg_url
+
+def format_changes(changes):
+    """Format the changes into a readable message"""
+    if not changes:
+        return "No dependency versions were changed"
+
+    def short_hash(hash):
+        """Return first 7 characters of commit hash"""
+        return hash[:7]
+
+    lines = []
+    updates = [c for c in changes if c['type'] == 'update']
+    additions = [c for c in changes if c['type'] == 'add']
+    removals = [c for c in changes if c['type'] == 'remove']
+
+    if updates:
+        lines.append("The following dependencies were updated:")
+        for change in updates:
+            url = get_github_diff_url(change['url'], change['old_version'], change['new_version'])
+            lines.append(f"* {change['dependency']}: {short_hash(change['old_version'])} → {short_hash(change['new_version'])} [[GitHub link]]({url})")
+
+    if additions:
+        if lines: lines.append("")
+        lines.append("The following dependencies were added:")
+        for change in additions:
+            url = get_github_diff_url(change['url'], None, change['version'])
+            lines.append(f"* {change['dependency']}: {short_hash(change['version'])} [[GitHub link]]({url})")
```

Comment:
```suggestion
            lines.append(f"* {change['dependency']}: {short_hash(change['version'])} [GitHub link]({url})")
```
I do not really know how python strings work, but are the double square brackets needed here?

Reply:
The double brackets should render as `[Github link]` which I think is aesthetically somewhat pleasing.